src/itensor.jl

```diff
@@ -1996,6 +1996,21 @@ function indpairs(T::ITensor; plev::Pair{Int,Int}=0 => 1, tags::Pair=ts"" => ts"
   return is_first .=> is_last
 end
 
+"""
+    ishermitian(T::ITensor; eps = 1E-10)
+
+Test whether an ITensor is a Hermitian operator,
+up to a numerical tolerance. To be considered an
+operator, an ITensor must have matching pairs
+of indices with prime level 0 and 1.
+
+# Optional Keyword Arguments
+- `eps = 1E-10`: the numerical tolerance to use 
+"""
+function ishermitian(T::ITensor; eps=1E-10)
```

Comment:
Let's use `atol` here to match Julia's `isapprox`:
```julia
help?> isapprox
search: isapprox

  isapprox(x, y; atol::Real=0, rtol::Real=atol>0 ? 0 : √eps, nans::Bool=false[, norm::Function])


  Inexact equality comparison. Two numbers compare equal if their relative distance or their absolute distance is within
  tolerance bounds: isapprox returns true if norm(x-y) <= max(atol, rtol*max(norm(x), norm(y))). The default atol is zero and the
  default rtol depends on the types of x and y. The keyword argument nans determines whether or not NaN values are considered
  equal (defaults to false).
```
That could anticipate using `rtol` for a relative tolerance, which could make more sense here anyway since it should be in some way relative to the norm of the `ITensor`.

Reply:
Yes, I was wondering if there was a lower-level thing we should be using here, sort of like how the \approx operator we use in tests calls a Base function.

Reply:
Actually we could just use:
```julia
ishermitian(T::ITensor; kwargs...) = isapprox(T, swapprime(dag(T), 0 => 1); kwargs...)
```

Additionally, this reminds me that I want to define:
```julia
transpose(T::ITensor) = swapprime(T, 0 => 1)
adjoint(T::ITensor) = transpose(dag(T))
```
as a shorthand so we don't have to write `swapprime(T, 0 => 1)` all the time.

Unfortunately the `adjoint` definition clashes with the definition `adjoint(T::ITensor) = prime(T)` to get the notation `T' == prime(T)` (see https://github.com/JuliaLang/julia/issues/44521), but we could add `transpose(T::ITensor)` right now.

Reply:
I just went ahead and implemented it using `isapprox` internally. By the way, we can use all this work on `expect` as a warmup of some of the improvements you mentioned for `correlation_matrix`, including using `isapprox` there for some of the Hermiticity tests etc.

Reply:
Ok we thought of the same thing - I'll go ahead and define transpose too, yes. Later we could make `transpose` more agnostic about the other prime level being 1, but since we use that convention so widely I think it's ok for now.

Reply:
Yes exactly. I've wanted to come up with a good standard for interfaces with operator-like ITensors which default to prime levels of 0 and 1, but then you can more generally choose the left and right indices with prime levels and tags.

One place I've done that is the `tr(::ITensor)` function: https://github.com/ITensor/ITensors.jl/blob/10d870155ef78ee6c1257ea07f9c1571a651c3bc/src/itensor.jl#L2002 but I think there could be some things to improve about the design there. Ideally we would have a common code pattern that could be shared across all functions that treat ITensors as operators (i.e. matricizing an ITensor), such as `tr`, `tranpose`, `adjoint`, `eigen`, `exp`, etc.

Reply:
Maybe we could chat about this more next week since it's a good design question. By the way, seeing this `tr` function makes me think I should update the Code Example about how to take the trace of an ITensor. (I could leave the existing material there in some form, but really the `tr` function is the proper way not mainly because it's a function but because `tr` should only be formally taken on indices which are identical in meaning).